diff -urNp orig/app/config/gimpbaseconfig.c new/app/config/gimpbaseconfig.c
--- orig/app/config/gimpbaseconfig.c	2009-07-20 22:20:50.000000000 +0200
+++ new/app/config/gimpbaseconfig.c	2016-08-31 14:49:17.817861634 +0200
@@ -100,6 +100,8 @@ gimp_base_config_class_init (GimpBaseCon
   num_processors = num_processors * 2;
 #endif
 
+  num_processors = MIN (num_processors, GIMP_MAX_NUM_THREADS);
+
   GIMP_CONFIG_INSTALL_PROP_UINT (object_class, PROP_NUM_PROCESSORS,
                                  "num-processors", NUM_PROCESSORS_BLURB,
                                  1, GIMP_MAX_NUM_THREADS, num_processors,
